api/samlacs.go

```diff
@@ -148,7 +154,23 @@ func (a *API) SAMLACS(w http.ResponseWriter, r *http.Request) error {
 		}
 	}
 
-	// TODO: fetch new metadata if possible when validUntil < time.Now() or cacheDuration
+	var samlMetadataModified bool
+	samlMetadataModified = false
+	if *ssoProvider.SAMLProvider.MetadataURL != "" && IsMetadataStale(idpMetadata, ssoProvider.SAMLProvider) {
+		rawMetadata, err := fetchSAMLMetadata(ctx, *ssoProvider.SAMLProvider.MetadataURL)
+		if err != nil {
+			// Fail silently but raise warning and continue with existing metadata
+			logentry := log.WithField("sso_provider_id", ssoProvider.ID.String())
+			logentry = logentry.WithField("expires_in", time.Until(idpMetadata.ValidUntil).String())
+			logentry = logentry.WithField("valid_until", idpMetadata.ValidUntil)
+			logentry = logentry.WithField("error", err)
```

Comment:
I think there was something like `logentry.Error()` for this purpose? Can you check?

Reply:
Hm maybe `WithError` ? I'll go with that